lib/Dialect/TritonGPU/Transforms/DotSlicing.cpp

```diff
@@ -123,17 +121,24 @@ struct TritonAMDGPUDotSlicingPass
       sliceOffsets.push_back(0);
     }
 
-    auto viewPtr = builder.create<ttg::ViewSliceOp>(
-        dotOp.getLoc(),
-        RankedTensorType::get(sliceSizes, ptrTensorType.getElementType(),
-                              ptrTensorType.getEncoding()),
-        ptrTensor, ValueRange({}), ValueRange({}), ValueRange({}), sliceOffsets,
-        sliceSizes, sliceStrides);
-
     // Begin with the load instruction and proceed to slice the operations
     // along the execution path of the dotOperand.
     IRMapping mapping;
-    mapping.map(ptrTensor, viewPtr);
+    ttg::ViewSliceOp viewPtr;
+    for (auto i = 0; i < firstOpToSlice->getOperands().size(); i++) {
+      auto arg = firstOpToSlice->getOperand(i);
+      if (auto tensorType = arg.getType().cast<RankedTensorType>()) {
```

Comment:
cast will fire assert if arg type is not `RankedTensorType`.

I think you mean to use `dyn_cast` instead of `cast`:
```suggestion
      if (auto tensorType = arg.getType().dyn_cast<RankedTensorType>()) {
```

P.s. I am not sure if any load arguments could be non tensor type. At least I've never saw such a case.